crates/optimism/rpc/src/error.rs

```diff
@@ -93,6 +97,27 @@ impl TryFrom<InvalidTransaction> for OpInvalidTransactionError {
     }
 }
 
+/// 4337 related error type
+#[derive(Debug, thiserror::Error)]
+pub enum Op4337Error {
+    /// Transaction conditional cost exceeded maximum allowed
+    #[error("conditional cost exceeded maximum allowed")]
+    ConditionalCostExceeded,
+    /// Invalid conditional parameters
+    #[error("invalid conditional parameters")]
+    InvalidCondition,
+}
+
+impl From<Op4337Error> for jsonrpsee_types::error::ErrorObject<'static> {
+    fn from(err: Op4337Error) -> Self {
+        jsonrpsee_types::error::ErrorObject::owned(
+            INTERNAL_ERROR_CODE,
+            err.to_string(),
+            None::<String>,
```

Comment:
I think we can return INVALID_PARAMS here?

Reply:
makes more sense, done